independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/DefineClassVisibleURLClassLoader.java

```diff
@@ -10,11 +12,48 @@
  */
 public class DefineClassVisibleURLClassLoader extends URLClassLoader {
 
+    private static final List<String> KNOWN_LEAKING_PACKAGES_FROM_PARENT_CL = Arrays.asList(
+            "org.apache.commons.lang3", // commons-lang3 used internally by gradle
+            "com.google.common" // guava added by maven internals
+    );
+
     public DefineClassVisibleURLClassLoader(URL[] urls, ClassLoader parent) {
         super(urls, parent);
     }
 
     public Class<?> visibleDefineClass(String name, byte[] b, int off, int len) throws ClassFormatError {
         return super.defineClass(name, b, off, len);
     }
+
+    public Class<?> loadClass(String name) throws ClassNotFoundException {
+        boolean lookup = false;
+        for (String prefix : KNOWN_LEAKING_PACKAGES_FROM_PARENT_CL) {
+            if (name.startsWith(prefix)) {
+                lookup = true;
+                break;
+            }
+        }
+
+        // The parent classloader can introduce plenty of build system dependencies (like commons-lang3 or guava)
+        // Which could be incompatible with what the user dependencies specify
+        // The temporary solution is to not delegate to parent class loader for dependencies that are known
+        // to leak in from the build system.
+        // The solution is pretty bad and should be views as temporary until a proper solution
+        // (like the one proposed here: https://github.com/quarkusio/quarkus/issues/770)
+        if (lookup) {
+            Class<?> loadedClass = findLoadedClass(name);
```

Comment:
I think this might be problematic. What is Gradle has already loaded some classes from guava? findLoadedClass will return the one loaded by the base class loader, but any classes that have not been loaded yet will be loaded by this class loader, breaking any package protected access and loading different versions.

Reply:
I am assuming that would only happen if some plugin used in the build depended on Guava and was unshaded. It could probably happen, but I am assuming it won't be very likely.

Reply:
Why would it be in the build classpath if it is not being used?

Reply:
Guava sneaks in (or at least used to sneak in) because it was a dependency of the maven resolver, or some other library we use in boostrap.